src/main/java/cam72cam/immersiverailroading/registry/DefinitionManager.java

```diff
@@ -78,98 +79,174 @@ public static void initDefinitions() throws IOException {
         definitions = new LinkedHashMap<>();
         tracks = new LinkedHashMap<>();
 
-        Set<String> defTypes = jsonLoaders.keySet();
+        initModels();
+        initModelHeightMaps();
+        initTracks();
+    }
 
-        List<String> blacklist = new ArrayList<>();
+    public static EntityRollingStockDefinition getDefinition(String defID) {
+        return definitions.get(defID);
+    }
 
-        Identifier blacklist_json = new Identifier(ImmersiveRailroading.MODID, "rolling_stock/blacklist.json");
+    public static Collection<EntityRollingStockDefinition> getDefinitions() {
+        return definitions.values();
+    }
 
-        List<InputStream> inputs = blacklist_json.getResourceStreamAll();
+    public static Set<String> getDefinitionNames() {
+        return definitions.keySet();
+    }
+
+    public static Collection<TrackDefinition> getTracks() {
+        return tracks.values();
+    }
+
+    public static List<String> getTrackIDs() {
```

Comment:
Could you move these functions back down in the file so they don't show up in the diff?